src/main/java/cqltoelm/parsers/MATCQL2ELMListener.java

```diff
@@ -414,13 +414,15 @@ private Element resolve(String identifier, TranslatedLibrary library) {
         // updated again if need be.
         if (element instanceof IncludeDef) {
             IncludeDef def = (IncludeDef) element;
-            libraries.add(def.getPath() + "-" + def.getVersion() + "|" + def.getLocalIdentifier());
             graph.addEdge(currentContext, def.getPath() + "-" + def.getVersion() + "|" + def.getLocalIdentifier());
             libraryAccessor = def;
-            try {
-                parseChildLibraries(def);
-            } catch (IOException e) {
-                e.printStackTrace();
+            if (!libraries.contains(def.getPath() + "-" + def.getVersion() + "|" + def.getLocalIdentifier())) {
+                libraries.add(def.getPath() + "-" + def.getVersion() + "|" + def.getLocalIdentifier());
+                try {
+                    parseChildLibraries(def);
```

Comment:
```suggestion
                try {
                    parseChildLibraries(def);
                    libraries.add(def.getPath() + "-" + def.getVersion() + "|" + def.getLocalIdentifier());
```
How about moving the `add` inside the try/catch block, right after `parseChildLibraries` to ensure we're only adding the library if the method is successfully. Though, we don't do anything different in the error case

Reply:
Yup it doesn't matter

Reply:
Yes, pls.